ui/src/util/basePaths.spec.ts

```diff
@@ -5,34 +5,59 @@ import {
 } from "./basePaths";
 
 vi.mock("./basePaths", async () => {
-  window.base = "/example";
+  const basePath = "/example/ui";
+  // Can't use the setBase function here as this mock gets hoisted to the top of
+  // the file so gets executed before setBase exists.
+  const base = document.createElement("base");
+  base.setAttribute("href", basePath);
+  document.body.appendChild(base);
   const actual = await vi.importActual("./basePaths");
   return {
     ...actual,
-    basePath: "/example/ui/",
-    apiBasePath: "/example/api/v0/",
+    basePath,
   };
 });
 
+const cleanBase = () =>
+  document.querySelectorAll("base").forEach((base) => {
+    document.body.removeChild(base);
+  });
+
+const setBase = (href: string) => {
+  const base = document.createElement("base");
+  base.setAttribute("href", href);
+  document.body.appendChild(base);
+};
+
+// Clean up the base created by the mock above.
+cleanBase();
+
 describe("calculateBasePath", () => {
+  afterEach(() => {
+    cleanBase();
+  });
+
   it("resolves with ui path", () => {
-    window.base = "/test/";
+    setBase("/test/");
     const result = calculateBasePath();
     expect(result).toBe("/test/");
   });
 
   it("resolves with ui path without trailing slash", () => {
-    window.base = "/test";
+    setBase("/test");
     const result = calculateBasePath();
     expect(result).toBe("/test/");
   });
 
-  it("resolves with root path if the base is not provided", () => {
-    if (window.base) {
-      delete window.base;
-    }
+  //   it("handles full URL", () => {
+  //     setBase("http://example.com/test/");
+  //     const result = calculateBasePath();
+  //     expect(result).toBe("/test/");
+  //   });
```

Comment:
This might need to be removed or uncommented.

Reply:
Oops! Thanks.